### PR TITLE
🐛 Source Postgres: allow handling of java sql date

### PR DESCRIPTION
**Pull request by shrodingers**
## What
Had an issue on a Postgres CDC source, where I have some Postgres date columns, and on the initial load, date object types were `Java.sql.Date`, which weren't handled properly. This caused a warning to be prompted for every row, and the date format conversion to fall on the default string converter.

## How
Added a handler for `Java.sql.Date` type that can be returned by Debeezium, instead of a repetitive warning causing unreadable logs. 

Allow for a specific handler and not the default one.

## 🚨 User Impact 🚨
Should be none, since it just adds a handler instead of the default one for debeezium based sources

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))<
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
thanks @shrodingers! I requested to the team review the contribution during this week.

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java [hunk 1]
```diff
@@ -42,7 +43,10 @@ public static String convertDate(final Object input) {
     } else if (input instanceof Number) {
       return DataTypeUtils.toISO8601String(
           new Timestamp(((Number) input).longValue()).toLocalDateTime());
-    } else if (input instanceof String) {
+    } else if (input instanceof Date) {
+      return DataTypeUtils.toISO8601String(input)
```
Comment (DoNotPanicUA) [inline airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java:47]: The `input` has `Object` type and can't be put as input parameter for the `DataTypeUtils.toISO8601String` without cast.

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java [hunk 2]
```diff
@@ -42,7 +43,10 @@ public static String convertDate(final Object input) {
     } else if (input instanceof Number) {
       return DataTypeUtils.toISO8601String(
           new Timestamp(((Number) input).longValue()).toLocalDateTime());
-    } else if (input instanceof String) {
+    } else if (input instanceof Date) {
```
Comment (DoNotPanicUA) [inline airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java:46]: Please specify the case which you are trying to fix with steps to reproduce.
I've tested your change using CDC MySql Source and there is no any issues with `Date` type. 
More over the class `java.sql.Date` is not used there.
All columns with `Date` type processed inside the Debezium as `class java.time.LocalDate`. Example from data type test:
Screenshot from 2022-01-20 14-20-27

**Review by DoNotPanicUA — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java (hunk 2 above)
Comment (shrodingers) [reply airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumConverterUtils.java:46]: I'm gonna update the PR description, im using CDC for PostgreSql for this case (thought it was the same case for all dbs thus my initial description). 
I had lots of warning while syncing columns with postgres's type Date looking like : 
Uncovered date class type 'java.sql.Date'. Use default converter. Which led me to this part of the code (only happened on the initial migration tho, not while consuming the slots).
I had a minimal setup to reproduce this, where i log the class name inside the `convertDate` function, and without this update i have the following log : 
image
Maybe it is a postgres only issue, or that i didn't understand how everything works around Debezium, but this is the reason i tried to submit this PR

**Review by DoNotPanicUA — CHANGES_REQUESTED** [review]
Hi @shrodingers ,
You PR looks good for me, but there are few small things left to do.
1) As this change will effect only Postgres processing and I will only incr. that version, please rename the PR from `Debeezium based sources` to `Source Postgres`.
2) The formatting check is a part of our build. It means that all changes should be in line with it. So, in order not to brake the master build, please make you changes in line with the changes in my technical PR #9753. (Note that sequence of imports also counts)
All tests are fine, so after that we can merge your PR and I will bump the version.

**Comment by shrodingers** [comment]
Hi @DoNotPanicUA, thanks for your patience in this review !
- Just changed the PR title, as it indeed only affected Postgres sources
- Merged your technical Pr inside mine in order to update it with airbyte's master and fix the formatting issues.
I'll remain available if there is anything else needed here

**Review by DoNotPanicUA — APPROVED** [review]
👍

**Comment by DoNotPanicUA** [comment]
Hi @shrodingers,
Please be informed that `0.4.3` Postgres version is published and now can be used.
Thank you for your input into our project ;)

**Pull request merged**